signals.go

```diff
@@ -45,6 +46,10 @@ func serverSignalHandler(listener net.Listener, statusListener net.Listener, sto
 			switch sig {
 			case syscall.SIGTERM:
 				logger.Printf("received SIGTERM, stopping listener")
+				time.AfterFunc(5*time.Minute, func() {
```

Comment:
Could we hoist this value out to a flag or something?  connections might take longer than 5 minutes, so we may want graceful shutdown to take arbitrarily long, depending on usecase